packages/type-utils/src/isTypeReadonly.ts

```diff
@@ -31,23 +36,65 @@ export const readonlynessOptionsSchema = {
     treatMethodsAsReadonly: {
       type: 'boolean',
     },
+    allowlist: {
+      type: 'array',
+      items: typeAllowListItemSchema,
+    },
   },
 };
 
 export const readonlynessOptionsDefaults: ReadonlynessOptions = {
   treatMethodsAsReadonly: false,
+  allowlist: [],
 };
 
 function hasSymbol(node: ts.Node): node is ts.Node & { symbol: ts.Symbol } {
   return Object.prototype.hasOwnProperty.call(node, 'symbol');
 }
 
+function isTypeExcepted(
+  type: ts.Type,
+  program: ts.Program,
+  options: ReadonlynessOptions,
+): boolean {
+  const typeName = type.getSymbol()?.escapedName;
+  const matchingItems =
+    options.allowlist?.filter(item => item.typeName === typeName) ?? [];
+  for (const item of matchingItems) {
+    const declarationFiles =
+      type
+        .getSymbol()
+        ?.getDeclarations()
+        ?.map(declaration => declaration.getSourceFile()) ?? [];
+    for (const declaration of declarationFiles) {
+      if (
+        // A local type defined in the current package
+        (item.source === 'local' &&
+          declaration.fileName.startsWith(program.getCurrentDirectory())) ||
+        // A type from the default library
+        (item.source === 'default-lib' &&
+          program.isSourceFileDefaultLibrary(declaration)) ||
+        // A type from a specified third-party package
+        (item.source === 'package' &&
+          (declaration.fileName.includes('node_modules/' + item.package) ||
+            declaration.fileName.includes(
+              'node_modules/@types/' + item.package,
+            )))
```

Comment:
this should probably be
`` `node_modules/${item.package}/` ``.

Without the trailing slash, `react` would match `node_modules/react-foo`